forc-pkg/src/pkg.rs

```diff
@@ -905,7 +1029,7 @@ fn dep_to_source(pkg_path: &Path, dep: &Dependency) -> Result<Source> {
         Dependency::Detailed(ref det) => match (&det.path, &det.version, &det.git) {
             (Some(relative_path), _, _) => {
                 let path = pkg_path.join(relative_path);
-                Source::Path(path)
+                Source::Path(fs::canonicalize(path)?)
```

Comment:
We can insert relative paths to the `Forc.toml` in the background they are also stored like that meaning the following:

Adding std as a dependency from `context_testing_abi` would result in  a path as the following:

`/Users/kayagokalp/sway/test/src/e2e_vm_tests/test_programs/should_pass/require_contract_deployment/caller_context_test/../../test_abis/context_testing_abi/../../../../../../../sway-lib-std`

which is basically 

`/Users/kayagokalp/sway/sway-lib-std`

Before this change, this was detected as a change which is actually not so I converted these paths to absolute paths before storing them as a pkg field.